contracts/Claims.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.7.5;
+
+// @notice Used to create Radicle Link identity claims
+contract Claims {
+    /// @notice Emitted on every Radicle Link identity claim
+    /// @param addr The account address which made the claim
+    event Claimed(address indexed addr);
+
+    /// @notice Creates a new claim of a Radicle Link identity.
+    /// Every new claim invalidates previous ones made with the same account.
+    /// The claims have no expiration date and don't need to be renewed.
+    /// @param version The version of `root` encoding, currently supported values:
+    /// - `1` - `root` is a SHA-1 root hash. The excess high bytes are zeros, e.g. for hash
+    /// `fb3102b74d7254eed7f18a31a3ba1ea946bb1a99` the passed `root` is
+    /// `000000000000000000000000fb3102b74d7254eed7f18a31a3ba1ea946bb1a99`
+    /// - `2` - `root` is a SHA-256 root hash
+    /// @param root The claimed Radicle Link identity root.
+    /// To revoke a claim without creating a new one, pass root `0`,
+    /// which is guaranteed to not match any existing identity.
+    function claim(uint8 version, uint256 root) public {
```

Comment:
For future compatibility, do we want to add a `byte32` as the third parameter? This would allow additional information to be anchored potentially.

Reply:
I would also swap the parameter order, as the `root` is the thing being claimed.

Reply:
> do we want to add a byte32 as the third parameter?

That's a good point. You've made me reconsidered it and now I think that we should go one step further and converge into this:

```
function claim(uint8 version, bytes payload) public {
```

- For `version == 1` `payload` must be 20 bytes long and hold the SHA-1 root hash.
- For `version == 2` `payload` must be 32 bytes long and hold the SHA-256 root hash.
- And so on

It's provides extreme elasticity without making the current usage any more complex. The gas overhead is minuscule, it's only extra 32 bytes in the transaction data. What do you think @cloudhead @NunoAlexandre ?

Reply:
Sounds good to me!

Reply:
Added. Also switched `version` from `uint8` to `uint256`, every calldata parameter takes multiples of 32 bytes anyway.

Reply:
Yup, I like this!